### ISSUE TITLE
[Feature Request] Can the IOS client app add the option to customize the moon and planet?

### ISSUE THREAD

**Issue by v808comd** [zerotier/ZeroTierOne#2351]
Can the IOS client app add the option to customize the moon and planet?

**Comment by FeatherStark** [comment]
Does Mac OS also support customization?

**Comment by joseph-henry** [comment]
It does not currently support this and it isn't on our immediate roadmap, but we aren't entirely opposed to the idea. I'll leave this ticket open until the next time we do significant work in the iOS app.

**Comment by mgolub2** [comment]
> Does Mac OS also support customization?

Yes, the files are in `/Library/Application\ Support/ZeroTier/One`
So for example, to replace the default planets with a custom planet file from a ztnet instance:
```bash
# Backups are cool(?)
cp  /Library/Application\ Support/ZeroTier/One/planet  /Library/Application\ Support/ZeroTier/One/planet.bak
curl https://ztnet.example.come/api/planet/ -o /Library/Application\ Support/ZeroTier/One/planet
```

**Comment by mgolub2** [comment]
> It does not currently support this and it isn't on our immediate roadmap, but we aren't entirely opposed to the idea. I'll leave this ticket open until the next time we do significant work in the iOS app.

I'd pay a (relatively small) amount for a subscription that lets you enable self hosting connections via the iOS app, if that's an issue stopping resources being allocated there.

**Comment by cheyue** [comment]
Mobile ios hope to have support

**Comment by ValiantHeartz** [comment]
Mobile ios hope to have support +1

**Comment by PetrusZ** [comment]
Mobile ios hope to have support +2

**Comment by XIUANGH** [comment]
Mobile ios hope to have support +3

**Comment by MayDomine** [comment]
Mobile ios hope to have support +4

**Comment by blimjoe** [comment]
Mobile ios hope to have support +5

There's jailbreak tweak for custom plant, but I still want official support.

**Comment by TohsakaOrz** [comment]
Mobile ios hope to have support +6

**Comment by Hononon** [comment]
Mobile ios hope to have support +7

**Comment by Monstry** [comment]
Mobile ios hope to have support +8

**Comment by suhli** [comment]
Mobile ios hope to have support +9

**Comment by kristianvmd** [comment]
Mobile ios hope to have support +10

**Comment by TPOP2000** [comment]
Mobile ios hope to have support +11

**Comment by nbb2025** [comment]
Mobile ios hope to have support +12

**Comment by Brunch-Life** [comment]
Mobile ios hope to have support +14

**Comment by hellopowermetal** [comment]
Mobile ios hope to have support +15

**Comment by noelfwfaw** [comment]
I discovered that this issue has existed for a long time, dating back to five years ago. Unfortunately, the development team has done nothing about it. I spent a significant amount of time setting up Moon and solving various details (such as port blocking and mapping) based on very limited tutorials, only to find it futile because iOS cannot use it. Moreover, if I have to control one computer using another, there’s no point in going through all this trouble. It’s time to abandon ZeroTier.

**Comment by mgolub2** [comment]
Are there other overlay networks with the same multi-network ability as zerotier? Tailscale requires a lot of hacking to support anything close to what zerotier does: https://jamesguthrie.ch/blog/multi-tailnet-unlocking-access-to-multiple-tailscale-networks/

**Comment by shadow1ng** [comment]
A simple method is to place the planet in the app Documents directory and allow iOS devices to access the app directory. Then we can replace the planet through a computer data cable

**Comment by tom-snow** [comment]
Only for jailbroken devices, and you should compile the dylib yourself.
https://github.com/lemon4ex/ZeroTieriOSFix

**Comment by rawvoid** [comment]
Mobile ios hope to have support +16

**Comment by zore-u** [comment]
mark

**Comment by mayongze** [comment]
Mobile ios hope to have support +17

**Comment by shady2k** [comment]
Mobile ios hope to have support +18

**Comment by LittleHeroZZZX** [comment]
Mobile ios hope to have support +19

**Comment by toantk238** [comment]
Mobile ios hope to have support +20

**Comment by dnvyrn** [comment]
Mobile ios hope to have support +21

**Comment by HaynesChennn** [comment]
Mobile ios hope to have support +22

**Comment by andreid1303** [comment]
Mobile ios hope to have support +23

**Comment by GuFFy12** [comment]
Mobile ios hope to have support +24

**Comment by JamesZBL** [comment]
Mobile ios hope to have support +25

**Comment by Hononon** [comment]
hey, guys! There is a way to add self-hosted plannet in IOS, as shown in [here](https://discuss.zerotier.com/t/add-custom-planets-to-new-1-14-2-ios-version/27895/11) and [here](https://blog.csdn.net/deephacking/article/details/151491767)

**Comment by XIUANGH** [comment]
> hey, guys! There is a way to add self-hosted plannet in IOS, as shown in [here](https://discuss.zerotier.com/t/add-custom-planets-to-new-1-14-2-ios-version/27895/11) and [here](https://blog.csdn.net/deephacking/article/details/151491767)

<img width="1013" height="586" alt="Image" src="https://github.com/user-attachments/assets/0bd2e652-a355-49ac-be02-a1238e27eabf" />

<img width="834" height="60" alt="Image" src="https://github.com/user-attachments/assets/d81ca955-9d0a-4f63-a006-7c180ab5ac94" />

<img width="640" height="559" alt="Image" src="https://github.com/user-attachments/assets/de39a843-0a61-4e93-86c8-510a60712b9f" />

I didn't seem to find the button for this gear, maybe it was removed in the 1.16 version? I used the URL to open zerotier but there was no response. Also, some people have reported that this feature is missing.
